ludwig/backend/ray.py

```diff
@@ -663,14 +689,28 @@ def eval_fn(
         # Deserialize the model (minus weights) from Plasma
         # Extract the weights from Plasma (without copying data)
         # Load the weights back into the model in-place on the current device (CPU)
-        model = distributed.replace_model_from_serialization(ray.get(model_ref))
+        model = distributed.replace_model_from_serialization(
+            ray.get(model_ref), optimization_stage=distributed_optimization_stage
+        )
         model = distributed.to_device(model)
 
-        # have to wrap here because we are passing into predictor directly.
+        # We have to wrap here because we are passing into predictor directly.
         # This is in contrast creating the predictor in the trainer class and
         # passing in the model post-wrap.
+        #
+        # For LLMs:
+        # If we are using DeepSpeed Stage 3, this wraps the model in DeepSpeed Zero Inference.
+        # For DeepSpeed Stage 2 and below, this loads the base model weights into the model.
+        # If the model was not trained using an adapter, the fine-tuned weights are loaded into the model.
+        # If the model was trained using an adapter, the base model weights are loaded into the model along with the
+        # adapter weights used at initialization. We will updated the state dict to use the weights from the
+        # fine-tuned adapter.
         dist_model = distributed.prepare_for_inference(model)
+        if adapter_ref and (distributed_optimization_stage and distributed_optimization_stage <= 2):
```

Comment:
This can be pushed into the DistributedStrategy, again, so we don't need to couple everything to DeepSpeed.